src/app/services/ws2.service.ts

```diff
@@ -0,0 +1,103 @@
+import { Inject, Injectable } from '@angular/core';
+import { Router } from '@angular/router';
+import { UUID } from 'angular2-uuid';
+import { environment } from 'environments/environment';
+import { EMPTY, Observable, of } from 'rxjs';
+import {
+  filter, map, share, switchMap, take,
+} from 'rxjs/operators';
+import { webSocket, WebSocketSubject } from 'rxjs/webSocket';
+import { IncomingApiMessageType, OutgoingApiMessageType } from 'app/enums/api-message-type.enum';
+import { WINDOW } from 'app/helpers/window.helper';
+import { ApiDirectory, ApiMethod } from 'app/interfaces/api-directory.interface';
+import { ApiEventDirectory, LogsEventDirectory } from 'app/interfaces/api-event-directory.interface';
+import { ApiEvent, IncomingWebsocketMessage, ResultMessage } from 'app/interfaces/api-message.interface';
+
+@Injectable()
+export class WebsocketService2 {
+  subject$: WebSocketSubject<unknown>;
+  private readonly eventSubscriptions: Map<string, Observable<unknown>> = new Map();
+  constructor(
+    protected router: Router,
+    @Inject(WINDOW) protected window: Window,
+  ) {
+    this.subject$ = webSocket(
+      (this.window.location.protocol === 'https:' ? 'wss://' : 'ws://')
+      + environment.remote + '/websocket',
+    );
+    this.send({
+      msg: OutgoingApiMessageType.Connect,
+      version: '1',
+      support: ['1'],
+    });
+
+    this.subject$.subscribe({
+      complete: () => {
+        this.router.navigate(['/sessions/signin']);
+      },
+    });
+    this.ping();
+  }
+
+  ping(): void {
+    this.subject$.next({ msg: OutgoingApiMessageType.Ping, id: UUID.UUID() });
+    setTimeout(() => this.ping(), 20 * 1000);
```

Comment:
Hard to imagine where this would be an issue, but we should store timeout and clear it on destroy just for sake of good practice.